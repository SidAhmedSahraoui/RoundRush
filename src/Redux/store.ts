import { Action, ThunkAction, configureStore } from "@reduxjs/toolkit";

import auth from "./auth/auth-slice";


export const store = configureStore({
  reducer: {
    auth
  },
  devTools: true,
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
