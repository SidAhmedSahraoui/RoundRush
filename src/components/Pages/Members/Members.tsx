import React, { Fragment } from "react";
import useStyles from "./style";
import Oval from "../../../img/Oval.svg"
import { v4 as uuidv4 } from "uuid"
import { uProps } from "../../../types/types";

const Members: React.FC<uProps> = ({users}) => {
  const classes = useStyles();


  return (
    <>
      <main className={classes.main}>
        <h1>Users on plan</h1>
        <table className="table text-left">
          <thead>
            <tr>
              <th style={{width: "50%"}} scope="col">Name</th>
              <th style={{width: "0px" , padding: "0px"}}  scope="col"></th>
              <th style={{width: "40%" , paddingLeft: "35px"}} scope="col">Email</th>
              <th style={{width: "10%"}} scope="col">Status</th>
              <th style={{width: "0px", padding: "0px"}} scope="col"></th>
            </tr>
          </thead>
          <tbody>
            {users.map((user) => (
                <Fragment key={uuidv4()} >
                    <tr className={!user.active ? "inact" : ""}>
                        <td className="name">
                            <img width="20px" height="20px" src={Oval} alt="" />
                            {user.name}
                        </td>
                        <td className="owner">
                            {user.owner ? <button>Owner</button> : null}
                        </td>
                        <td className="email">
                            {user.email}
                        </td>
                        <td>
                            {user.active ? <h5 className="active">Active</h5> : <h5 className="inactive">Inactive</h5>}
                        </td>
                        <td className="points">
                            ...
                        </td>
                    </tr>
                    <span className="sb"></span>
                </Fragment>
            ))}
          </tbody>
        </table>
      </main>
    </>
  );
};
export default Members;
