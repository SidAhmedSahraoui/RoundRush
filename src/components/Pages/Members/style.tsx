import { createUseStyles } from "react-jss";

const useStyles = createUseStyles({
  main: {
    paddingTop: "12px",
    "& h1": {
      marginTop: "2px",
      fontFamily: "Rubik",
      fontSize: "24px",
      fontWeight: 400,
      textAlign: "left",
      color: "#31394E",
    },
    "& .btn": {
      width: "100%",
      display: "flex",
      justifyContent: "flex-end",
      alignItems: "center",
      marginTop:"24px"

    },
    "& .table": {
      width: "700px",
      marginTop: "40px",
      "& .head": {
        "& .line": {
          display: "flex",
          justifyContent: "flex-start",
          alignItems: "center",
          "& .col": {
            width: "20%",
            padding: "12px 55px 40px 10px",
            fontFamily: "Rubik",
            fontSize: "14px",
            fontWeight: 600,
            color: "rgba(49, 57, 78, 0.5)",
            textAlign: "left",
          },
        },
      },
      "& main": {
        "& .sb": {
          width: "100%",
          height: "2px",
          margin: "0px",
        },
        "& .inact": {
          backgroundColor: "#EAEBED",
        },
        "& div": {
          padding: "12px",
          display: "flex",
          justifyContent: "space-between",
          alignItems: "center",
          "& .name": {
            width: "170px",
            display: "flex",
            justifyContent: "flex-start",
            fontFamily: "Rubik",
            fontSize: "16px",
            fontWeight: 400,
            color: "#31394E",
            "& img": {
              marginRight: "26px",
            },
          },

          "& .email": {
            fontFamily: "Rubik",
            fontSize: "12px",
            fontWeight: 400,
            color: "#838895",
          },
          "& .points": {
            color: "#838895"
          },
          "& .active": {
            width: "50px",
            fontFamily: "Rubik",
            fontSize: "12px",
            fontWeight: 400,
            color: "#29C293",
          },
          "& .inactive": {
            width: "50px",
            fontFamily: "Rubik",
            fontSize: "12px",
            fontWeight: 400,
            color: "#FD5461",
          },
          "& .owner": {
            width:"90px",
            "& button": {
              width: "84px",
              height: "24px",
              textAlign: "center",
              fontFamily: "Rubik",
              fontSize: "12px",
              fontWeight: 400,
              border: "none",
              borderRadius: "6px",
              color: "#29C293",
              backgroundColor: "#BFEDDF",
            },
            
          },
        },
      },
    },
  },
});
export default useStyles;
