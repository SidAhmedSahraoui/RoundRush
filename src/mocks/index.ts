import { Task, Todo, Objective, userObject } from "../types/types";

export const tasks: Array<Task> = [
  {
    name: "LD Website",
    code: "LW",
    color: "#DB54FD",
  },
  {
    name: "LD Marketing",
    code: "LM",
    color: "#61D26F",
  },
  {
    name: "RoundRush",
    code: "R",
    color: "#1FAAE9",
  },
  {
    name: "RoundRush Website",
    code: "RW",
    color: "#4086E0",
  },
  {
    name: "DevOps",
    code: "DO",
    color: "#EEC301",
  },
  {
    name: "MicroServices",
    code: "MS",
    color: "#FD612C",
  },
  {
    name: "Deep Learning",
    code: "DL",
    color: "#FD612C",
  },
  {
    name: "SpringBoot",
    code: "SP",
    color: "#1FAAE9",
  },
  {
    name: "Java",
    code: "J",
    color: "#61D26F",
  },
];
export const todos: Array<Todo> = [
    {
        priorety: 4,
        type: "task",
        code: "VMS-20",
        title: "Certe, inquam, pertinax non intell...",
        parent: "LD Website",
        color: "#61D26F",
      },
      {
        priorety: 4,
        type: "task2",
        code: "VMS-11",
        title: "Ac hac et ac tristique luctus.",
        parent: "Roundrush",
        color: "#1FAAE9",
      },
      {
        priorety: 3,
        type: "macro",
        code: "VMS-28",
        title: "Magna bibendum nibh mi habitant.",
        parent: "Marketing",
        color: "#1FAAE9",
      },
      {
        priorety: 3,
        type: "task",
        code: "VMS-22",
        title: "Ac hac et ac tristique luctus.",
        parent: "LD Website",
        color: "#61D26F",
      },
      {
        priorety: 2,
        type: "task2",
        code: "VMS-11",
        title: "Varius ultricies vitae eget.",
        parent: "LD Website",
        color: "#7A6FEF",
      },
      {
        priorety: 1,
        type: "macro",
        code: "VMS-22",
        title: "Certe, inquam, pertinax non intell...",
        parent: "LD Website",
        color: "#7A6FEF",
      },
];
export const objs: Array<Objective> = [
    {
        name: "Notifications Center",
        priority: 4,
        status: false,
        date: "15/01/2020",
        check: "5/10",
      },
      {
        name: "App Mobile",
        priority: 3,
        status: false,
        date: "15/01/2020",
        check: "5/10",
      },
      {
        name: "DevOps",
        priority: 2,
        status: false,
        date: "15/01/2020",
        check: "5/10",
      },
      {
        name: "Kubernetes",
        priority: 1,
        status: false,
        date: "15/01/2020",
        check: "5/10",
      },
      {
        name: "Docker",
        priority: 4,
        status: false,
        date: "15/01/2020",
        check: "5/10",
      },
]

export const users: Array<userObject> = [
  {
    name: "Mario Panariello",
    owner: true,
    email: "mario.panariello@lastingdynamics.com",
    active: true,
  },
  {
    name: "Brandon Watson",
    owner: false,
    email: "mario.panariello@lastingdynamics.com",
    active: true,
  },
  {
    name: "Norma Copper",
    owner: false,
    email: "mario.panariello@lastingdynamics.com",
    active: false,
  },
  {
    name: "Mario Panariello",
    owner: false,
    email: "mario.panariello@lastingdynamics.com",
    active: false,
  },
  {
      name: "Mario Panariello",
      owner: true,
      email: "mario.panariello@lastingdynamics.com",
      active: true,
    }
]