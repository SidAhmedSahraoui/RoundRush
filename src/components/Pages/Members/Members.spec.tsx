/**
 * @jest-environment jsdom
 */

 import React from "react";
 import { cleanup, render } from "@testing-library/react";
 import { Provider } from "react-redux";
 import { BrowserRouter as Router } from "react-router-dom";
 import { store } from "../../../redux/store";
 import { users } from "../../../mocks";
import Members from "./Members";
 
 describe("Members", () => {
   afterEach(() => cleanup());
 
   test("Render Members with Data from Mock", () => {
     const view = render(
       <Provider store={store}>
         <Router>
           <Members users={users} />
         </Router>
       </Provider>
     );
 
     expect(view.container).toMatchSnapshot();
   });
 });
 