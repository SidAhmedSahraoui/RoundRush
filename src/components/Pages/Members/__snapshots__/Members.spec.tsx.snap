// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Members Render Members with Data from Mock 1`] = `
<div>
  <main
    class="main-0-2-1"
  >
    <h1>
      Users on plan
    </h1>
    <table
      class="table text-left"
    >
      <thead>
        <tr>
          <th
            scope="col"
            style="width: 50%;"
          >
            Name
          </th>
          <th
            scope="col"
            style="width: 0px; padding: 0px;"
          />
          <th
            scope="col"
            style="width: 40%; padding-left: 35px;"
          >
            Email
          </th>
          <th
            scope="col"
            style="width: 10%;"
          >
            Status
          </th>
          <th
            scope="col"
            style="width: 0px; padding: 0px;"
          />
        </tr>
      </thead>
      <tbody>
        <tr
          class=""
        >
          <td
            class="name"
          >
            <img
              alt=""
              height="20px"
              src="Oval.svg"
              width="20px"
            />
            Mario Panariello
          </td>
          <td
            class="owner"
          >
            <button>
              Owner
            </button>
          </td>
          <td
            class="email"
          >
            mario.panariello@lastingdynamics.com
          </td>
          <td>
            <h5
              class="active"
            >
              Active
            </h5>
          </td>
          <td
            class="points"
          >
            ...
          </td>
        </tr>
        <span
          class="sb"
        />
        <tr
          class=""
        >
          <td
            class="name"
          >
            <img
              alt=""
              height="20px"
              src="Oval.svg"
              width="20px"
            />
            Brandon Watson
          </td>
          <td
            class="owner"
          />
          <td
            class="email"
          >
            mario.panariello@lastingdynamics.com
          </td>
          <td>
            <h5
              class="active"
            >
              Active
            </h5>
          </td>
          <td
            class="points"
          >
            ...
          </td>
        </tr>
        <span
          class="sb"
        />
        <tr
          class="inact"
        >
          <td
            class="name"
          >
            <img
              alt=""
              height="20px"
              src="Oval.svg"
              width="20px"
            />
            Norma Copper
          </td>
          <td
            class="owner"
          />
          <td
            class="email"
          >
            mario.panariello@lastingdynamics.com
          </td>
          <td>
            <h5
              class="inactive"
            >
              Inactive
            </h5>
          </td>
          <td
            class="points"
          >
            ...
          </td>
        </tr>
        <span
          class="sb"
        />
        <tr
          class="inact"
        >
          <td
            class="name"
          >
            <img
              alt=""
              height="20px"
              src="Oval.svg"
              width="20px"
            />
            Mario Panariello
          </td>
          <td
            class="owner"
          />
          <td
            class="email"
          >
            mario.panariello@lastingdynamics.com
          </td>
          <td>
            <h5
              class="inactive"
            >
              Inactive
            </h5>
          </td>
          <td
            class="points"
          >
            ...
          </td>
        </tr>
        <span
          class="sb"
        />
        <tr
          class=""
        >
          <td
            class="name"
          >
            <img
              alt=""
              height="20px"
              src="Oval.svg"
              width="20px"
            />
            Mario Panariello
          </td>
          <td
            class="owner"
          >
            <button>
              Owner
            </button>
          </td>
          <td
            class="email"
          >
            mario.panariello@lastingdynamics.com
          </td>
          <td>
            <h5
              class="active"
            >
              Active
            </h5>
          </td>
          <td
            class="points"
          >
            ...
          </td>
        </tr>
        <span
          class="sb"
        />
      </tbody>
    </table>
  </main>
</div>
`;

exports[`Objectives Render Objectives with Data from Mock 1`] = `
<div>
  <main
    class="main-0-2-1"
  >
    <h1>
      Users on plan
    </h1>
    <table
      class="table text-left"
    >
      <thead>
        <tr>
          <th
            scope="col"
            style="width: 50%;"
          >
            Name
          </th>
          <th
            scope="col"
            style="width: 0px; padding: 0px;"
          />
          <th
            scope="col"
            style="width: 40%; padding-left: 35px;"
          >
            Email
          </th>
          <th
            scope="col"
            style="width: 10%;"
          >
            Status
          </th>
          <th
            scope="col"
            style="width: 0px; padding: 0px;"
          />
        </tr>
      </thead>
      <tbody>
        <tr
          class=""
        >
          <td
            class="name"
          >
            <img
              alt=""
              height="20px"
              src="Oval.svg"
              width="20px"
            />
            Mario Panariello
          </td>
          <td
            class="owner"
          >
            <button>
              Owner
            </button>
          </td>
          <td
            class="email"
          >
            mario.panariello@lastingdynamics.com
          </td>
          <td>
            <h5
              class="active"
            >
              Active
            </h5>
          </td>
          <td
            class="points"
          >
            ...
          </td>
        </tr>
        <span
          class="sb"
        />
        <tr
          class=""
        >
          <td
            class="name"
          >
            <img
              alt=""
              height="20px"
              src="Oval.svg"
              width="20px"
            />
            Brandon Watson
          </td>
          <td
            class="owner"
          />
          <td
            class="email"
          >
            mario.panariello@lastingdynamics.com
          </td>
          <td>
            <h5
              class="active"
            >
              Active
            </h5>
          </td>
          <td
            class="points"
          >
            ...
          </td>
        </tr>
        <span
          class="sb"
        />
        <tr
          class="inact"
        >
          <td
            class="name"
          >
            <img
              alt=""
              height="20px"
              src="Oval.svg"
              width="20px"
            />
            Norma Copper
          </td>
          <td
            class="owner"
          />
          <td
            class="email"
          >
            mario.panariello@lastingdynamics.com
          </td>
          <td>
            <h5
              class="inactive"
            >
              Inactive
            </h5>
          </td>
          <td
            class="points"
          >
            ...
          </td>
        </tr>
        <span
          class="sb"
        />
        <tr
          class="inact"
        >
          <td
            class="name"
          >
            <img
              alt=""
              height="20px"
              src="Oval.svg"
              width="20px"
            />
            Mario Panariello
          </td>
          <td
            class="owner"
          />
          <td
            class="email"
          >
            mario.panariello@lastingdynamics.com
          </td>
          <td>
            <h5
              class="inactive"
            >
              Inactive
            </h5>
          </td>
          <td
            class="points"
          >
            ...
          </td>
        </tr>
        <span
          class="sb"
        />
        <tr
          class=""
        >
          <td
            class="name"
          >
            <img
              alt=""
              height="20px"
              src="Oval.svg"
              width="20px"
            />
            Mario Panariello
          </td>
          <td
            class="owner"
          >
            <button>
              Owner
            </button>
          </td>
          <td
            class="email"
          >
            mario.panariello@lastingdynamics.com
          </td>
          <td>
            <h5
              class="active"
            >
              Active
            </h5>
          </td>
          <td
            class="points"
          >
            ...
          </td>
        </tr>
        <span
          class="sb"
        />
      </tbody>
    </table>
  </main>
</div>
`;
